client/collection.go

```diff
@@ -32,7 +34,7 @@ type CollectionDefinition struct {
 // Many functions on this object will interact with the underlying datastores.
 type Collection interface {
 	// Name returns the name of this collection.
-	Name() string
+	Name() immutable.Option[string]
 
 	// ID returns the ID of this Collection.
 	ID() uint32
```

Comment:
question: Will `ID` always be there even if `Name()` isn't? or should this also be optional?

Reply:
ID will always be there.